### PR TITLE
Change default startup behavior to not require ClickHouse

### PR DESCRIPTION
**Pull request by virajmehta**

### CODE REVIEW

File: gateway/src/main.rs [hunk 1]
```diff
@@ -19,16 +20,15 @@ static GLOBAL: MiMalloc = MiMalloc;
 async fn main() {
     // Set up logs and metrics
     observability::setup_logs();
-    let metrics_handle = observability::setup_metrics().expect_pretty("Failed to set up metrics");
+    let metrics_handle = observability::setup_metrics().ok_or_log();
```
Comment (GabrielBianconi) [inline gateway/src/main.rs:23]: This should still fail i think, nothing to do with ClickHouse. (And generally shouldn't even fail.)

File: gateway/src/main.rs [hunk 2]
```diff
@@ -46,11 +46,14 @@ async fn main() {
...
-        .route(
+        .with_state(app_state);
+
+    if let Some(metrics_handle) = metrics_handle {
+        router = router.route(
             "/metrics",
             get(move || std::future::ready(metrics_handle.render())),
         )
-        .with_state(app_state);
+    }
```
Comment (GabrielBianconi) [inline gateway/src/main.rs:56]: Metrics shouldn't fail because of clickhouse, no need?

File: gateway/src/gateway_util.rs [hunk 3]
```diff
@@ -17,24 +17,36 @@ pub struct AppStateData {
...
+        let clickhouse_url = std::env::var("CLICKHOUSE_URL")
+            .map_err(|_| Error::AppState {
                 message: "Missing environment variable CLICKHOUSE_URL".to_string(),
-            })?;
-
-            ClickHouseConnectionInfo::new(&clickhouse_url)?
-        };
+            })
+            .ok_or_log()
+            .unwrap_or_else(|| "".to_string());
```
Comment (GabrielBianconi) [inline gateway/src/gateway_util.rs:26]: It should not log if variable is missing but observability is disabled

Comment (GabrielBianconi) [reply gateway/src/gateway_util.rs:26]: Move this into `setup_clickhouse` in the `else` branch.

**Review by GabrielBianconi — CHANGES_REQUESTED** [review]

File: gateway/src/main.rs (hunk 2 above)
Comment (virajmehta) [reply gateway/src/main.rs:56]: I was trying to be as conservative as possible about keeping the gateway up but I guess it's OK

Comment (virajmehta) [reply gateway/src/main.rs:56]: After investigating, this only fails if you've already exported the metrics. OK to keep an expect

**Review by nikcaryo-super — APPROVED** [review]

**Pull request merged**